core/src/main/scala/za/co/absa/spline/core/harvester/componentCreators.scala

```diff
@@ -121,6 +122,7 @@ object ExpressionConverter {
       case map: Map[_, _] => asOption[Map[String, _]](for ((k, v) <- map; r <- recursion(v)) yield k.toString -> r)
       case seq: Traversable[_] => asOption(seq.flatMap(item => recursion(item)).toList)
       case _ if symbol.isModuleClass => Option(symbol.name.toString)
+      case row : GenericInternalRow => Option(row.toSeq(t.get.asInstanceOf[StructType]).flatMap(x => recursion(x)))
```

Comment:
Prefer using `Option.collect()` to avoid calling `get` with casting.
Also I'm not sure about `flatMap`. Shouldn't it be rather `map' with `orNull`?
```suggestion
case row: GenericInternalRow => t.collect({
    case st: StructType => row.toSeq(st).map(x => recursion(x).orNull)
})
```
Please add more unit tests to cover internal structures and structures with null-fields.